src/lib/OpenAI/Auth/BaseManifestAuth.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using Microsoft.OpenApi.ApiManifest.OpenAI.Auth;
+using System.Text.Json;
+
+namespace Microsoft.OpenApi.ApiManifest.OpenAI;
+
+public abstract class BaseManifestAuth
+{
+    public string? Type { get; set; }
+    public string? Instructions { get; set; }
+
+    public static BaseManifestAuth? Load(JsonElement value)
+    {
+        BaseManifestAuth? auth = null;
+
+        switch (value.GetProperty("type").GetString())
```

Comment:
to lower invariant maybe?

Reply:
yes, and I should move the property names to constants.